Geometry_Engine/Query/Area.cs

```diff
@@ -43,9 +43,9 @@ public static partial class Query
         [PreviousVersion("4.3", "BH.Engine.Geometry.Query.IArea(BH.oM.Geometry.IGeometry)")]
         public static double IArea(this IGeometry geometry, double tolerance = Tolerance.Distance)
         {
-            if (geometry == null)
+            if (geometry == null || tolerance == double.NaN)
```

Comment:
I do not think such construct will work for `double`, `double.IsNaN(tolerance)` is recommended, following [StackOverflow](https://stackoverflow.com/questions/558267/equality-with-double-nan):

> The [IEEE 754 floating point standard](https://en.wikipedia.org/wiki/IEEE_754) states that comparing NaN with NaN will always return false. If you must do this, use `Double.IsNaN()`.

But TBH, I would consider ignoring the check on `tolerance`, otherwise it would be good to rule out zero too etc. I think null checks are enough, good to hear others' opinion too though.